crates/biome_js_analyze/src/analyzers/nursery/use_node_import_protocol.rs

```diff
@@ -62,121 +55,91 @@ declare_rule! {
-                        // SAFETY: the list has one argument, checked by the if before
-                        let argument = arguments.iter().next().unwrap().ok()?;
-                        let argument = argument
-                            .as_any_js_expression()?
-                            .as_any_js_literal_expression()?
-                            .as_js_string_literal_expression()?
-                            .inner_string_text()
-                            .ok()?;
-                        if is_node_module_without_protocol(argument.text()) {
-                            return Some(argument.to_string());
-                        }
-                    }
-                }
-            }
-            AnyJsImportLike::JsImportCallExpression(import_call) => {
-                let arguments = import_call.arguments().ok()?.args();
-                if arguments.len() == 1 {
-                    // SAFETY: the list has one argument, checked by the if before
-                    let argument = arguments.iter().next().unwrap().ok()?;
-                    let argument = argument
-                        .as_any_js_expression()?
-                        .as_any_js_literal_expression()?
-                        .as_js_string_literal_expression()?
-                        .inner_string_text()
-                        .ok()?;
-                    if is_node_module_without_protocol(argument.text()) {
-                        return Some(argument.to_string());
-                    }
-                }
-            }
-        };
-
-        None
+        let module_name = ctx.query().module_name_token()?;
+        is_node_module_without_protocol(&inner_string_text(&module_name)).then_some(module_name)
     }
 
-    fn diagnostic(ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
-        let binding = ctx.query();
-
+    fn diagnostic(_: &RuleContext<Self>, module_name: &Self::State) -> Option<RuleDiagnostic> {
         Some(RuleDiagnostic::new(
             rule_category!(),
-            binding.range(),
+            module_name.text_trimmed_range(),
             markup! {
-                "Import from Node.js builtin module \""<Emphasis>{state}</Emphasis>"\" should use the \""<Emphasis>"node:"</Emphasis>"\" protocol."
+                "A Node.js builtin module should be imported with the "<Emphasis>"node:"</Emphasis>" protocol."
             },
         )
         .note(markup!{
             "Using the "<Emphasis>"node:"</Emphasis>" protocol is more explicit and signals that the imported module belongs to Node.js."
         }))
     }
 
-    fn action(ctx: &RuleContext<Self>, state: &Self::State) -> Option<JsRuleAction> {
-        let binding = ctx.query();
-        let new_import_str = format!("node:{}", state);
+    fn action(ctx: &RuleContext<Self>, module_name: &Self::State) -> Option<JsRuleAction> {
+        debug_assert!(module_name.kind() == JsSyntaxKind::JS_STRING_LITERAL);
```

Comment:
We usually want to add a message in the assertions. This message should hint a way to fix the issue in case the assertion fails